Terrafor 2.0.0rc1: Unable to retrieve schemas: failed to get schemas: terraform

I removed all traces of the preview extension and installed this new one. However, when opening a folder in VSCode, it looks like things fail in two ways:

1) 2020/06/04 19:21:39 schema_storage.go:116: error obtaining schemas: Unable to retrieve schemas: failed to get schemas: terraform (pid 9924) exited (code 1): exit status 1

2) stderr: "\x1b[31m\n\x1b[1m\x1b[31mError: \x1b[0m\x1b[0m\x1b[1mError obtaining Authorization Token from the Azure CLI: Error parsing json result from the Azure CLI: Error launching Azure CLI: exec: \"az\": executable file not found in %PATH%\x1b[0m\n\n\x1b[0m\x1b[0m\x1b[0m\n"

For item 1, I'm not sure what is going on.

For item 2, I have confirmed that 'az' is in PATH:

```
H:\>where az
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\az

H:\>echo %PATH%
C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Python27\;C:\Python27\Scripts;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\SafeNet\Authentication\SAC\x64;C:\Program Files\SafeNet\Authentication\SAC\x32;C:\Program Files (x86)\Enterprise Vault\EVClient\x64\;C:\Program Files\Citrix\System32\;C:\Program Files\Citrix\ICAService\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Go\bin;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\;C:\Program Files\Git\cmd;C:\Program Files\Common Files\microsoft shared\OFFICE16\;C:\Program Files\PowerShell\7\;C:\Users\XXXXXX\AppData\Local\Microsoft\WindowsApps;C:\Users\XXXXXX\AppData\Roaming\npm;C:\Users\XXXXXX\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\XXXXXX\go\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;

H:\>az --version
azure-cli                          2.7.0

command-modules-nspkg              2.0.3
core                               2.7.0
nspkg                              3.0.4
telemetry                          1.0.4
```

So I am not sure why terraform-ls is complaining if can't find the command, because from what I can tell it is there.

Log output below. I snipped out the contents of the terraform file. If needed, I can add it back in.

```
Launching language server: c:\Users\XXXXXX\.vscode\extensions\hashicorp.terraform-2.0.0-rc.1/lsp/terraform-ls serve
2020/06/04 19:21:37 service.go:63: Preparing new session ...
2020/06/04 19:21:37 langserver.go:79: Starting server (pid 128) ...
2020/06/04 19:21:37 server.go:469: Received 1 new requests
2020/06/04 19:21:37 server.go:165: Processing 1 requests
2020/06/04 19:21:37 server.go:223: Checking request for "initialize": {"processId":11788,"rootPath":"c:\\Scratch\\DataLab","rootUri":"file:///c%3A/Scratch/DataLab","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":

<snip some of log output>

2020/06/04 19:21:37 exec.go:182: Starting C:\ProgramData\chocolatey\bin\terraform.exe ["terraform" "version"] in "C:\\Users\\XXXXXX\\AppData\\Local\\Temp"...
2020/06/04 19:21:37 exec.go:148: Waiting for command to finish ...
2020/06/04 19:21:38 exec.go:174: terraform run (C:\ProgramData\chocolatey\bin\terraform.exe ["terraform" "version"], in "C:\\Users\\XXXXXX\\AppData\\Local\\Temp", pid 11372) finished with exit code 0
2020/06/04 19:21:38 initialize.go:90: Found compatible Terraform version (0.12.26) at C:\ProgramData\chocolatey\bin\terraform.exe
2020/06/04 19:21:38 schema_storage.go:334: Adding workspace for watching: "c:\\Scratch\\DataLab"
2020/06/04 19:21:38 watcher.go:88: Adding "c:\\Scratch\\DataLab\\.terraform\\plugins\\windows_amd64\\lock.json" for watching...
2020/06/04 19:21:38 schema_storage.go:124: Acquiring semaphore before retrieving schema for "c:\\Scratch\\DataLab" ...
2020/06/04 19:21:38 schema_storage.go:133: Retrieving schemas for "c:\\Scratch\\DataLab" ...
2020/06/04 19:21:38 exec.go:182: Starting C:\ProgramData\chocolatey\bin\terraform.exe ["terraform" "providers" "schema" "-json"] in "c:\\Scratch\\DataLab"...
2020/06/04 19:21:38 schema_storage.go:310: Watching for plugin changes ...
2020/06/04 19:21:38 rpc_logger.go:50: Response to "initialize" (ID 0): {"capabilities":{"textDocumentSync":{"openClose":true,"change":2},"completionProvider":{},"documentFormattingProvider":true}}
2020/06/04 19:21:38 server.go:204: Completed 1 requests [630.443ms elapsed]
2020/06/04 19:21:38 server.go:469: Received 1 new requests
2020/06/04 19:21:38 server.go:469: Received 1 new requests
2020/06/04 19:21:38 server.go:165: Processing 1 requests
2020/06/04 19:21:38 server.go:223: Checking request for "initialized": {}
2020/06/04 19:21:38 server.go:165: Processing 1 requests
2020/06/04 19:21:38 server.go:223: Checking request for "textDocument/didOpen": {"textDocument":{"uri":"file:///c%3A/Scratch/DataLab/01-setup.tf","languageId":"terraform","version":1,"text":"

<snip terraform code>

"}}
2020/06/04 19:21:38 exec.go:148: Waiting for command to finish ...
**2020/06/04 19:21:39 schema_storage.go:116: error obtaining schemas: Unable to retrieve schemas: failed to get schemas: terraform (pid 9924) exited (code 1): exit status 1**
stdout: ""
stderr: "\x1b[31m\n\x1b[1m\x1b[31mError: \x1b[0m\x1b[0m\x1b[1mError obtaining Authorization Token from the Azure CLI: Error parsing json result from the Azure CLI: Error launching Azure CLI: exec: \"az\": executable file not found in %PATH%\x1b[0m\n\n\x1b[0m\x1b[0m\x1b[0m\n"
```

--- a/internal/terraform/exec/exec.go
+++ b/internal/terraform/exec/exec.go
@@ -21,15 +21,11 @@ import (
 
 var defaultExecTimeout = 30 * time.Second
 
-// Environment variables to pass through to Terraform
-var passthroughEnvVars = []string{
-	// This allows Terraform to find custom-built providers
-	"HOME", "USER", "USERPROFILE",
-	// This allows Terraform to create crash log in the desired temp directory
-	// os.TempDir would otherwise fall back to C:\Windows on Windows
-	// which has no write permissions for non-admins
-	"TMPDIR", "TMP", "TEMP",
-}
+// We pass through all variables, but longer term we'll need to reflect
+// that some variables might be workspace/directory specific
+// and passing through these may be dangerous once the LS
+// starts to execute commands which can mutate the state
+var passthroughEnvVars = os.Environ()
 
 // cmdCtxFunc allows mocking of Terraform in tests while retaining
 // ability to pass context for timeout/cancellation

